packages/web-components/src/components/ic-select/ic-select.tsx

```diff
@@ -305,7 +305,8 @@ export class Select {
       }
     }
 
-    if (this.searchable) {
+    if (this.searchable && this.value) {
+      // Only set if value not null - prevents whole input value being cleared when edited
       this.searchableSelectInputValue =
         this.getLabelFromValue(this.currValue as string) ||
```

Comment:
using `as <type>` is an anti-pattern, maybe just construct a string? I think the warnings will ignore that, then.
 
```js
this.getLabelFromValue(new String(this.currValue))
```

Reply:
I was just wondering why this is anti-pattern / whether you have any references which explain why `as <type>` shouldn't be used? We have used it many times throughout our codebase, and I have seen it used in a lot of other code online

Also where are the warnings you are seeing? :)

Reply:
We sometimes used `as <type>` in webpack because the function parameter had a different type than the one passed as an argument.

Without `as <type>` it made the IDE (or CLI) typechecking emit an error, so that's why I assumed that you used (`as string`). 

If the expected argument passed to a function differs than the one being passed (you override the typechecking), it seems more like you're making typescript not typecheck that specific code, if that makes sense?

Reply:
I've found that [some sources](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html) seem to recommend against using `String`

Reply:
Try without the `new` keyword.

Reply:
That worked - thanks!

As for my previous comment about it not being recommended, I have looked at some other sources and opinions seem to be mixed, so happy to just go with it especially as we have sometimes used `String` elsewhere in our code

Reply:
Good stuff 🤓